promql/engine.go

```diff
@@ -720,16 +751,25 @@ func (ev *evaluator) rangeEval(f func([]Value, *EvalNodeHelper) Vector, exprs ..
 		enh.ts = ts
 		result := f(args, enh)
 		enh.out = result[:0] // Reuse result vector.
+
+		// In some cases we only have the result of 'f' without having tracked any samples
+		// in the above for loop that built the input vectors.
+		ev.currentSamples += len(result)
```

Comment:
keep this?

Reply:
Yes, this will be part of the output

What you ultimately want is the original tempNumSamples plus all of these in ev.currentSamples when you return